app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,52 @@
+class OrdersController < ApplicationController
+  before_action :find_order_id, only: %i[show complete update]
+
+  def update
+    if @order.update(order_params)
+      flash[:notice] = 'Order status updated successfully.'
+      OrderNotifierMailer.order_status_notification(@order, @order.status).deliver_now
+    else
+      flash[:alert] = 'Failed to update order status.'
+    end
+    redirect_to user_path(@order.user)
+  end
+
+  def new
+    @order = Order.new
+  end
+
+  def create
+    @cart = current_user.cart
+    @order = current_user.orders.build(order_params)
+    @order.cart = @cart
+    if @order.save
+      redirect_to order_path(@order)
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def show; end
+
+  def complete
```

Comment:
Can you please explain the working of this action.

Reply:
Complete action is used when the user provides all details like the shipping address, then clicks on the complete order button this action triggers and sends an email to the user that their order is placed.